### PR TITLE
Refactor datetime-picker

### PR DESCRIPTION
**Pull request by goremikins**

### CODE REVIEW

File: packages/elements/src/datetime-picker/index.ts [hunk 1]
```diff
@@ -880,44 +694,12 @@ export class DatetimePicker extends ControlElement implements MultiValue {
...
-    }
-
-    if (path.includes(this.iconEl)) {
-      this.toggleOpened();
-    }
-    else if (!this.inputTriggerDisabled) {
-      this.setOpened(true);
-    }
+  private onButtonTap (event: TapEvent): void {
+    this.setOpened(true);
```
Comment (Sarin-Udompanish) [inline packages/elements/src/datetime-picker/index.ts:667]: The popup will not close when click on the button again.

File: packages/elements/src/datetime-picker/index.ts [hunk 2]
```diff
@@ -880,44 +694,12 @@ export class DatetimePicker extends ControlElement implements MultiValue {
...
-      return; /* popup is managed separately */
-    }
-
-    if (path.includes(this.iconEl)) {
-      this.toggleOpened();
-    }
-    else if (!this.inputTriggerDisabled) {
-      this.setOpened(true);
-    }
+  private onButtonTap (event: TapEvent): void {
```
Comment (Sarin-Udompanish) [inline packages/elements/src/datetime-picker/index.ts]: Unused variable ( event )

File: packages/elements/src/datetime-picker/index.ts [hunk 3]
```diff
@@ -546,6 +547,31 @@ export class DatetimePicker extends ControlElement implements MultiValue {
...
+  private filterInvalidViews (views: string[]): string[] {
+    const filtered = [];
+
+    // views must match in duplex mode
+    if (views.length !== (this.isDuplex() ? 2 : 1)) {
+      return [];
+    }
+
+    for (let i = 0; i < views.length && filtered.length <= 2; i += 1) {
+      const view = views[0];
```
Comment (Sarin-Udompanish) [inline packages/elements/src/datetime-picker/index.ts]: Should it be `const view = views[i]` ?

Comment (goremikins) [reply packages/elements/src/datetime-picker/index.ts]: Done

**Pull request merged**